src/Algebra/Ordered/Structures.agda

```diff
@@ -0,0 +1,205 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Ordered algebraic structures (not packed up with sets, orders,
+-- operations, etc.)
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via
+-- `Algebra.Ordered`.
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Ordered.Structures where
+
+open import Algebra.Core
+open import Algebra.Definitions
+open import Algebra.Structures
+open import Algebra.Ordered.Definitions using (Monotonic₂)
+open import Data.Product using (proj₁; proj₂)
+open import Level using (_⊔_)
+open import Relation.Binary.Core using (Rel; _⇒_)
+open import Relation.Binary.Definitions using (Transitive)
+open import Relation.Binary.Structures using (IsPreorder; IsPartialOrder)
+
+------------------------------------------------------------------------
+-- Preordered structures
+
+module _ {a ℓ₁ ℓ₂} {A : Set a}
+  (_≈_ : Rel A ℓ₁)  -- The underlying equality.
+  (_∼_ : Rel A ℓ₂)  -- The preorder.
+  where
+
+  -- Preordered semigroups (prosemigroups)
+
+  record IsProsemigroup (_∙_ : Op₂ A) : Set (a ⊔ ℓ₁ ⊔ ℓ₂) where
```

Comment:
I know nothing about this field of algebra so take this with a pinch of salt, but should there also be a `IsPromagma`?

Reply:
I'm not an expert either. Honestly, I hadn't heard of magmas before seeing them defined in the stdlib. 😅  As I said, this PR is derived from some old code I had, and in that project I just mechanized those structures I needed. But I'm happy to add promagmans and pomagmas if you think they are useful.

Reply:
Hmm I think best to play it safe and add them. If I've learnt anything from maintaining this library is that there's always someone out there who wants full generality.

Reply:
Added promagmas and pomagmas now.